File: generator/src/main/groovy/ch/bailu/gtk/tag/DocumentTag.java
package ch.bailu.gtk.tag;

import ch.bailu.gtk.builder.BuilderInterface;

public class DocumentTag extends Tag {

    private final BuilderInterface builder;

    public DocumentTag(BuilderInterface builder) {
        super(null);
        this.builder = builder;
    }

    @Override
    public BuilderInterface getBuilder() {
        return builder;
    }

    @Override
    public Tag getChild(String name, String prefix) {
        if ("repository".equals(name)) {
            return new RepositoryTag(this);
        }
        return ignore();
    }
}

File: generator/src/main/kotlin/ch/bailu/gtk/parser/tag/DocumentTag.kt
package ch.bailu.gtk.parser.tag

import ch.bailu.gtk.NamespaceConfig
import ch.bailu.gtk.builder.BuilderInterface

class DocumentTag(private val builder: BuilderInterface,
                  private val namespaceConfig: NamespaceConfig
): Tag() {

    override fun getBuilder(): BuilderInterface {
        return builder
    }

    override fun getNamespaceConfig(): NamespaceConfig {
        return namespaceConfig
    }

    override fun getChild(name: String): TagWithParent {
        if ("repository" == name) {
            return RepositoryTag(this)
        }
        return ignore()
    }

    override fun getParent(): Tag {
        return this
    }
}
